Firestore/Source/Model/FSTMutation.h

```diff
@@ -45,7 +44,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (instancetype)init NS_UNAVAILABLE;
 - (instancetype)initWithVersion:(model::SnapshotVersion)version
-               transformResults:(NSArray<FSTFieldValue *> *_Nullable)transformResults
+               transformResults:(absl::optional<std::vector<model::FieldValue>>)transformResults
```

Comment:
I'm not convinced this is the right type here. This could be an empty vector and not harm anything. I've chosen to preserve this optional vector for now because it matches what came before but I think we could revisit.